tq42/client.py

```diff
@@ -177,38 +214,51 @@ def login(self):
                 data=data_token,
                 headers=self.environment.headers,
             )
-            json_token = response_token.json()
+            response_json = response_token.json()
             # If we received an access token, print it and break out of the loop
-            if "access_token" in json_token:
-                access_token = json_token["access_token"]
-
-                save_location = utils.save_token(
-                    service_name="access_token",
-                    backup_save_path=self.token_file_path,
-                    token=access_token,
-                )
-
-                print(
-                    f"Authentication is successful, access token is saved in: {save_location}."
-                )
-
-                env_set = environment_default_set(client=self)
-                print(env_set)
-
-            if "refresh_token" in json_token:
-                refresh_token = json_token["refresh_token"]
-                utils.save_token(
-                    service_name="refresh_token",
-                    backup_save_path=self.refresh_token_file_path,
-                    token=refresh_token,
-                )
-                current_datetime = datetime.now()
-                file_handling.write_to_file(self.timestamp_file_path, current_datetime)
+
+            if self.save_access_token(response_json) and self.save_refresh_token(
```

Comment:
executing a function that is called `save_` inside an if statement is an absolute no-go for me.

Please refactor this logic to actually indicate more what is happening.

Reply:
May you explain further? The logic that is used here is equivalent to the previous one. How do you prefer the refactor logic to flow?